plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1424,6 +1453,13 @@ func NewVolumeNodePredicate(pvInfo PersistentVolumeInfo, pvcInfo PersistentVolum
 }
 
 func (c *VolumeNodeChecker) predicate(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+
+	// Don't run this predicate if volume topology scheduling is enabled
+	// TODO msau: how do I properly remove this predicate???
```

Comment:
@bsalamat I want to remove/replace this predicate that I added in 1.7 as an alpha feature, but it seems like the scheduler has some backwards compatibility policy.  Could you advise on what I should do here?

Reply:
What is the specific issue you are seeing when you remove the predicate?

Reply:
The compatibility test fails because the predicate existed in 1.7 but not anymore in 1.9.  It seems like we cannot ever remove predicates.

Reply:
If the compatibility tests use a scheduler policy config and they don't update it in their upgrade process to remove the deleted predicate, they tests may fail. We must ensure that the compatibility test policy config is updated during the upgrade process.

Reply:
In the compatibility [test](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/algorithmprovider/defaults/compatibility_test.go#L318), it says that we cannot change the json or else we'll break backwards compatibility.  Are you saying that since the predicate I added in 1.7 is an alpha feature and does not have upgrade guarantees, then it is an allowed exception to this policy?

Or is there some sort of scheduler upgrade process that I can modify?

Reply:
I couldn't remove the predicate from defaults.go, unless I want to remove the predicate from the 1.7 compatibility test config.  It seems like the compatibility test is testing against the registered default predicates.